joomla/joomla-cms
# [4.0] Fields/editor/codemirror as custom element

Pull Request for Issue # .

### Summary of Changes
Move the inline jQuery dependant code to custom elements/native ES6 javascript


### Testing Instructions
Install this branch, change the default editor to codemirror and check it's functionality (F10 and all the highlighting still works)

<img width="1125" alt="screen shot 2018-03-12 at 01 11 37" src="https://user-images.githubusercontent.com/3889375/37259748-05ebd0aa-2593-11e8-8e86-00048087e9c7.png">
<img width="1471" alt="screen shot 2018-03-12 at 01 11 58" src="https://user-images.githubusercontent.com/3889375/37259752-0b8e3e76-2593-11e8-8d26-75799d0f7087.png">


### Few notes
- All the related scripts (codemirror, addons the custom element itself) are lazy loaded
- thus it makes this editor PWA ready
- We're using Promises (the polyfill for the webcomponents cover the older browsers)
- we're using requestAnimationFrame (https://caniuse.com/#feat=requestanimationframe) supported even in IE11 (wow 😮)
- the code remains flexible as the options can be changed in runtime (even better than the old var approach)

### Documentation Changes Required
Yes
